homeassistant/components/bluesound/media_player.py

```diff
@@ -619,22 +613,36 @@ def extra_state_attributes(self) -> dict[str, Any] | None:
 
     def rebuild_bluesound_group(self) -> list[str]:
         """Rebuild the list of entities in speaker group."""
-        if self._group_name is None:
+        if self.sync_status.master is None and self.sync_status.slaves is None:
             return []
 
-        device_group = self._group_name.split("+")
+        player_entities: list[BluesoundPlayer] = self.hass.data[DATA_BLUESOUND]
+
+        leader_sync_status: SyncStatus | None = None
+        match self.sync_status.master:
+            case None:
+                leader_sync_status = self.sync_status
+            case paired_player:
+                leader_sync_status_list = [
+                    x.sync_status
+                    for x in player_entities
+                    if x.sync_status.id == f"{paired_player.ip}:{paired_player.port}"
+                ]
+                if leader_sync_status_list:
+                    leader_sync_status = leader_sync_status_list[0]
```

Comment:
```suggestion
                for x in player_entities:
                    if x.sync_status.id == f"{paired_player.ip}:{paired_player.port}":
                        leader_sync_status = x.sync_status
                        break
```

We don't need to iterate over all entries if we only need the first matching one